analyze_model.py

```diff
@@ -100,3 +100,15 @@ def generate_heatmap(data, row_labels, col_labels, filename, output_dir, verbose
         fig.savefig(output_dir + filename)
         if verbose:
             plt.show()
+
+    @staticmethod
+    def generate_heatmap_from_results(self, heatmap_name: str, results: dict, save_results_path: str, verbose=False):
+        data = [value.values() for value in results.values()]
+        row_labels = results.keys()
+        col_labels = results["start"].keys()
```

Comment:
Kind of hacky, but if we never change `SaveModelAtEpochsCallback.on_train_begin` this should be fine